src/index.js

```diff
@@ -33,6 +40,10 @@ export default function <ResultFn: (...Array<any>) => mixed>(resultFn: ResultFn,
     return lastResult;
   };
 
+  defineProperty(result, 'length', resultFn.length);
+
+  defineProperty(result, 'name', `memoized_${resultFn.name || 'one'}`);
```

Comment:
I am a little worried that this might slow things done a bit. I am not sure if the effort is worth the sugar. Would you mind taking a look at the impact?

Reply:
Perhaps we could make them both getters! As you said - worth testing! 

Thanks @theKashey!